pkg/sfu/downtrack.go

```diff
@@ -1344,17 +1421,15 @@ func (m *Munger) UpdateAndGetSnTs(extPkt *buffer.ExtPacket) (uint16, uint32, err
 	// if there are gaps, record it in missing sequence number cache
 	diff := extPkt.Packet.SequenceNumber - m.highestIncomingSN
 	if diff > 1 {
+		var lossStartSN, lossEndSN int
+		lossStartSN = int(m.highestIncomingSN) + 1
 		if extPkt.Packet.SequenceNumber > m.highestIncomingSN {
-			for lostSN := m.highestIncomingSN + 1; lostSN < extPkt.Packet.SequenceNumber; lostSN++ {
-				m.missingSNs[lostSN] = m.snOffset
-			}
+			lossEndSN = int(extPkt.Packet.SequenceNumber) - 1
 		} else {
-			for lostSN := m.highestIncomingSN + 1; lostSN <= uint16(buffer.MaxSN-1); lostSN++ {
-				m.missingSNs[lostSN] = m.snOffset
-			}
-			for lostSN := uint16(0); lostSN < extPkt.Packet.SequenceNumber; lostSN++ {
-				m.missingSNs[lostSN] = m.snOffset
-			}
+			lossEndSN = int(extPkt.Packet.SequenceNumber) - 1 + buffer.MaxSN
+		}
+		for lostSN := lossStartSN; lostSN <= lossEndSN; lostSN++ {
+			m.missingSNs[uint16(lostSN&0xffff)] = m.snOffset
```

Comment:
Cleaning this up to use `int` range as `uint16` could have caused trouble in wrap around case check.

Reply:
Maybe we can add extSN field to extPacket, that can avoid check seq wrap everywhere.